python_modules/dagster/dagster/core/object_store.py

```diff
@@ -116,13 +116,20 @@ def get_valid_target_path(base_dir, paths):
 
 
 class FileSystemObjectStore(ObjectStore):
-    def __init__(self, run_id, types_to_register=None):
+    def __init__(self, run_id, types_to_register=None, base_dir=None):
         self.run_id = check.str_param(run_id, 'run_id')
         self.storage_mode = RunStorageMode.FILESYSTEM
-        self.root = get_run_files_directory(run_id)
+        self._base_dir = check.opt_nonempty_str_param(
```

Comment:
should we be ensuring that this directory exists here?

Reply:
probably